Port mav_msgs to ROS 2

srmainwaring opened the pull request:
This is a port of `mav_comm` to ROS 2, required for a downstream port of `terrain-navigation`.

@Jaeyoung-Lim could you please create a `ros2` branch and I'll update the PR to target that.

### Related

- https://github.com/ethz-asl/terrain-navigation/pull/20

srmainwaring committed:
ros2: port mav_comm to ros2 humble

- Port mav_msgs to ros2 humble

Signed-off-by: Rhys Mainwaring <rhys.mainwaring@me.com>

- Update additional includes.

Signed-off-by: Rhys Mainwaring <rhys.mainwaring@me.com>

- Update conversions and package deps.

Signed-off-by: Rhys Mainwaring <rhys.mainwaring@me.com>

- Update package and cmake files.

Signed-off-by: Rhys Mainwaring <rhys.mainwaring@me.com>

- Port mav_planning_msgs to ros2 humble

Signed-off-by: Rhys Mainwaring <rhys.mainwaring@me.com>

- Port mav_state_machine_msgs to ros2 humble

Signed-off-by: Rhys Mainwaring <rhys.mainwaring@me.com>

- Port mav_system_msgs to ros2 humble

Signed-off-by: Rhys Mainwaring <rhys.mainwaring@me.com>

- Port meta-package mav_comm to ros2 humble

Signed-off-by: Rhys Mainwaring <rhys.mainwaring@me.com>

Jaeyoung-Lim commented:
> @Jaeyoung-Lim could you please create a ros2 branch and I'll update the PR to target that.

Done!

srmainwaring commented:
> Done!

Thanks @Jaeyoung-Lim. I don't have access to the Jenkins job to see what needs resolving / updating for CI. Probably need a ROS 2 enabled machine but I can't tell from the info in the repo.

Let me know what you'd like completed to get this merged as it's a pre-requisite for the terrain-navigation port. It's not too much work at this stage to complete the port for the remaining packages if that is preferred.

**Update: 17 Nov**

@Jaeyoung-Lim - I've converted the remaining packages / msgs to ROS 2 and squashed everything down to a single commit. If you are ok with changes could you please approve and merge into the `ros2` branch.
